nix/devcontainer/default.nix

```diff
@@ -0,0 +1,100 @@
+{ name ? "devcontainer"
+, tag ? null
+, extraContents ? [ ]
+, extraCommands ? ""
+, dockerTools
+, bashInteractive
+, cacert
+, closureInfo
+, coreutils
+, curl
+, direnv
+, findutils
+, gcc-unwrapped
+, git
+, glibc
+, gnugrep
+, gnused
+, gnutar
+, gzip
+, iana-etc
+, iproute
+, less
+, lib
+, nix
+, openssh
+, procps
+, shadow
+, xz
+, which
+}:
+let
+  image = dockerTools.buildImage {
```

Comment:
Mind using [buildLayeredImage](https://nixos.org/manual/nixpkgs/stable/#ssec-pkgs-dockerTools-buildLayeredImage) instead?

Reply:
Sure, seems like it's a drop in replacement and maybe slightly faster?

Reply:
We have a enough stuff in the image that this seems to max out the layers pretty fast, so I don't know how much it will help :thinking:

Reply:
the max number of layers is configurable - i think it should help a lot with caching compared to the standard `buildImage` - that's the theory anyhow :)

Reply:
Sorry, commented in the wrong place: for some reason switching to `buildLayeredImage` builds an image that missing some of the files? :(

Reply:
Giving up on this for now, seems like it might be a bug in `buildLayeredImage` regarding string contexts.